addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3915,50 +3915,66 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
      * @param {function} computeShapeData function to compute the new shape data.
      */
     _handlePreviewState(previewMode, computeShapeData) {
+        const target = this.$target[0];
+        const insertShapeContainer = newContainer => {
+            const shapeContainer = target.querySelector(':scope > .o_we_shape');
+            if (shapeContainer) {
+                shapeContainer.remove();
+            }
+            if (newContainer) {
+                const preShapeLayerElement = this._getLastPreShapeLayerElement();
+                if (preShapeLayerElement) {
+                    preShapeLayerElement.after(newContainer);
+                } else {
+                    target.prepend(newContainer);
```

Comment:
Experimental function, I would not use it. Was like that before, but we can chance anyway.